UpdateRCConfiguration/handler.ts

```diff
@@ -0,0 +1,152 @@
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessNoContent
+} from "@pagopa/ts-commons/lib/responses";
+import { pipe } from "fp-ts/lib/function";
+import {
+  RCConfiguration,
+  RCConfigurationModel
+} from "@pagopa/io-functions-commons/dist/src/models/rc_configuration";
+import { NonEmptyString, Ulid } from "@pagopa/ts-commons/lib/strings";
+import { NewRCConfigurationPublic } from "../generated/definitions/NewRCConfigurationPublic";
+import { RequiredUserIdMiddleware } from "../middlewares/required_headers_middleware";
+import { makeNewRCConfigurationWithConfigurationId } from "../utils/mappers";
+
+export const isUserAllowedToUpdateConfiguration = (
+  userId: NonEmptyString
+): ((
+  configuration: RCConfiguration
+) => TE.TaskEither<IResponseErrorForbiddenNotAuthorized, RCConfiguration>) =>
+  TE.fromPredicate(
+    configuration => configuration.userId === userId,
+    () => ResponseErrorForbiddenNotAuthorized
+  );
+
+export const handleUpsert = (rccModel: RCConfigurationModel) => (
+  newRCConfiguration: RCConfiguration
+): TE.TaskEither<IResponseErrorInternal, IResponseSuccessNoContent> =>
+  pipe(
+    rccModel.upsert(newRCConfiguration),
+    TE.mapLeft(e =>
+      ResponseErrorInternal(
+        `Something went wrong trying to upsert the configuration: ${e}`
+      )
+    ),
+    TE.map(ResponseSuccessNoContent)
+  );
+
+export const handleEmptyConfiguration = (
+  maybeConfiguration: O.Option<RCConfiguration>
+): TE.TaskEither<IResponseErrorNotFound, RCConfiguration> =>
+  pipe(
+    maybeConfiguration,
+    TE.fromOption(() =>
+      ResponseErrorNotFound(
+        `Configuration not found`,
+        `Cannot find any remote-content configuration`
+      )
+    )
+  );
+
+export const handleGetLastRCConfigurationVersion = (
+  rccModel: RCConfigurationModel,
+  configurationId: Ulid
+): TE.TaskEither<IResponseErrorInternal, O.Option<RCConfiguration>> =>
+  pipe(
+    rccModel.findLastVersionByModelId([configurationId]),
+    TE.mapLeft(e =>
+      ResponseErrorInternal(
+        `Something went wrong trying to retrieve the configuration: ${e}`
+      )
+    )
+  );
+
+interface IHandlerParameter {
+  readonly configurationId: Ulid;
+  readonly newRCConfiguration: NewRCConfigurationPublic;
+  readonly userId: NonEmptyString;
+}
+
+interface IUpdateRCConfigurationHandlerParameter {
+  readonly rccModel: RCConfigurationModel;
+}
+
+type UpdateRCConfigurationHandlerReturnType = (
+  parameter: IHandlerParameter
+) => Promise<
+  | IResponseSuccessNoContent
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+  | IResponseErrorForbiddenNotAuthorized
+>;
+
+type UpdateRCConfigurationHandler = (
+  parameter: IUpdateRCConfigurationHandlerParameter
+) => UpdateRCConfigurationHandlerReturnType;
+
+export const updateRCConfigurationHandler: UpdateRCConfigurationHandler = ({
+  rccModel
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
why was this rule disabled?

Reply:
I think that since we define the handler as a `UpdateRCConfigurationHandler` it's to avoid rewriting the same things defined in that type. Some linting rules sometimes make you rewrite a lot of duplicated code.

Reply:
The linter complains because the return type of the function is not specified even if the `updateRCConfigurationHandler` is specified as `UpdateRCConfigurationHandler`, we have 2 options here:

* add the disable for the rule;
* Separate the types for the parameter in input and the type in output.

What's your take?